### ISSUE TITLE
https://online.visualstudio.com/environments page logged me out after a period of inactivity

### ISSUE THREAD

**Issue by fpintos** [MicrosoftDocs/vsonline#322]
I've opened https://online.visualstudio.com/environments this morning and logged in. I the tab (in Edge chromium) and moved on to other tabs.

After some time (1 hour, give or take) I returned to that particular tab in the browser, and the page jumped to the AAD logout, and subsequently asked me to login again.

I suppose tab discard (edge://discards/) might have something to do with it, but the logout/re-login was not expected.

**Comment by jkeech** [comment]
@legomushroom recently made a change to improve security which automatically logs you out after 30 minutes. I think this is standard practice for many Microsoft services including the Azure portal which requires you to reauthorize after a period of inactivity.

**Comment by legomushroom** [comment]
That is correct, we have to log you out after 30 minutes of inactivity for security reasons. We also might look into an opt-in option to explicitly disable such behaviour.

**Comment by olivermt** [comment]
This has an implentation error. It doesn't just "log out out". The way this is implemented currently it prompts full AAD system wide logout in the browser you are in by showing the "choose an account to log out" (if you have multiple).

This is a pretty annoying anti-pattern, because the standard behavior across our other services is that you just have to click to reauthenticate with your AAD session. That means that our muscle memory is trained to click the correct account. This results (at least five times so far) in me logging out my account by accident, invalidating a bunch of tabs every time I browse by the vsonline timed out window :)

So I suggest that you still keep logging people out, but do not redirect them to the /logout aad page, because that should not be necessary. Maybe redirect back to the vsonline main page?

**Comment by lostintangent** [comment]
This suggestion seems reasonable to me. 

@legomushroom Thougts?

**Comment by nikmd23** [comment]
I’d also really love the “Keep me logged in” button. I often do research in other tabs for 30+ mins and want to come right back to the position I was in.

**Comment by davide-rota** [comment]
Hello, a couple of considerations:
-VS Online doesn't really log you out, it just brings you to the page where you can choose which account to sign-out. If I don't select anything and just reload the VS Online url I'm back in with SSO.
- Consider that security best practices doesn't normally recommend to log out a user without an explicit request or specific reason (see when Azure AD normally revokes a session https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens#token-revocation )
- Device SSO (PRT Token) is still there and would let you SSO in again, so this risks to become just an annoyance for discontinued work sessions.
